### PR TITLE
objects to csv export: check for the dotted object path

### LINKED ISSUE #809: adminExport: field names starting by object paths are wrongly exported in csv file
For example, a field named `household.personsDidTrips` in a survey where there is `household.persons` object will export the content of this field in the household_persons file instead of the interviews one. We should add the '.' at the end of the object path to get the object's paths instead of just the name

### PR DESCRIPTION
**Pull request by tahini**
fixes #809

To avoid simple paths being added to an object file because they have the same prefix (for example `household.personsDidSomething` in the `household_persons` file, we look for the object path with a '.' (eg. `household.persons.` in the path.

Also add unit tests for the exportAllToCsvByObject export and fix array of objects being exported as en empty field with the root object.

Add comments in some places as necessary and remove the `arrayPaths` field as it is not used or necessary.

### CODE REVIEW

**Comment by tahini** [comment]
@kaligrafy j'ai regardé quelques enquêtes (od_qc, longue distance phase 1 et 2) et l'export semble bien équivalent, fix en plus.

Pourrais-tu regarder le test pour voir si tu penses à d'autres cas à tester.

Le coverage est pas trop pire, mais je pourrais me forcer plus j'imagine ;-)

File                        | % Stmts | % Branch | % Funcs | % Lines |    Uncovered Line #s                                    
----------------------------|---------|----------|---------|---------|------------------------------------------------                                                         
  exportAllToCsvByObject.ts |   85.87 |    76.34 |   70.73 |   85.49 | 55-76,96,101,140,173,216-217,235,402-407,415-416,478,515,526,543-555

File: packages/evolution-backend/src/services/adminExport/__tests__/exportAllToCsvByObject.test.ts [hunk 1]
```diff
@@ -0,0 +1,439 @@
...
+        }));
+        expect(visitedPlacesRows[1]).toEqual(expect.objectContaining({
+            _interviewUuid: interviewData.uuid,
+            _parentUuid: person2Uuid,
+            _uuid: visitedPlace1P2Uuid,
+            name: interviewData.responses.household.persons[person2Uuid].visitedPlaces[visitedPlace1P2Uuid].name,
+            'geography.type': interviewData.responses.household.persons[person2Uuid].visitedPlaces[visitedPlace1P2Uuid].geography.type,
+            'geography.properties.action': interviewData.responses.household.persons[person2Uuid].visitedPlaces[visitedPlace1P2Uuid].geography.properties.action,
+            'geography.geometry.type': interviewData.responses.household.persons[person2Uuid].visitedPlaces[visitedPlace1P2Uuid].geography.geometry.type,
+            'geography.geometry.lat': String(interviewData.responses.household.persons[person2Uuid].visitedPlaces[visitedPlace1P2Uuid].geography.geometry.coordinates[1]),
```
Comment (tahini) [inline packages/evolution-backend/src/services/adminExport/__tests__/exportAllToCsvByObject.test.ts:328]: Si j'enlève le objectContaining, je vois qu'il y a aussi un export de `geography.geometry.coordinates`. Est-ce qu'on le laisse? Ou .lat et .lon doivent remplacer ce champs dans le nouveau format des fichiers?

Comment (kaligrafy) [reply packages/evolution-backend/src/services/adminExport/__tests__/exportAllToCsvByObject.test.ts:328]: ce serait mieux d'enlever geography.geometry.coordinates
J'avais la flemme de le faire parce que c'était compliqué...

Comment (tahini) [reply packages/evolution-backend/src/services/adminExport/__tests__/exportAllToCsvByObject.test.ts:328]: C'est fait, il manquait juste une `else` en fait...

**Review by kaligrafy — APPROVED** [review]
